tests/core/test_utils.py

```diff
@@ -149,3 +154,31 @@ def test_json_params_sagemaker():
         assert len(include_collections) == 2
         assert hook_params["export_tensorboard"] == True
         assert hook_params["tensorboard_dir"] == sim.tensorboard_dir
+
+
+def test_is_first_process():
+    for _ in range(10):
+        helper_test_is_first_process()
+
+
+def helper_test_is_first_process():
+    path = "/tmp/first_process"
```

Comment:
please use TempDir instead of non-portable, harcoded paths.

Reply:
Done